### DISCUSSION TITLE
Use JLink or SWD to provide MicroPython firmware for RT1062 chip？

### DISCUSSION THREAD

**Discussion by allan203** [micropython/micropython#15410] [m.iMX RT / Teensy 4.x]
Hello everyone, 
I made a PCB board using the RT1062 chip based on TEENSY4.1, but I didn't design a similar burning chip（MKL02ZXXX） for TEENSY4.1. 
Can I use jlink or swd to burn the MicroPython firmware?
Thank you very much！！

**Comment by robert-hh** [comment]
Sure, you can do that. I use SWD for all RT boards except the Teensy. For development, the loader chip of the Teensy is more an obstacle than a support, since it blocks the jlink pins. 
There are as well tools by NXP a serial load method. That method is used in the instructions for the Olimex board. But using Jlink is more convenient.

**Comment by allan203** [comment]
Dear Robert, 
I am so happy to receive your message.
I am a beginner in programming. If you can help me describe the details of the two methods you mentioned, I will be very grateful!
Thank you again！



&nbsp;
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "micropython/micropython"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月5日(星期五) 下午2:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [micropython/micropython] Use JLink or SWD to provide MicroPython firmware for RT1062 chip？ (Discussion #15410)





 
Sure, you can do that. I use SWD for all RT boards except the Teensy. For development, the loader chip of the Teensy is more an obstacle than a support, since it blocks the jlink pins.
 There are as well tools by NXP a serial load method. That method is used in the instructions for the Olimex board. But using Jlink is more convenient.

**Comment by robert-hh** [comment]
I use the Segger JLink adapters for loading firmware and for debugging. The EDU models are sufficient and affordable. Segger offers the Ozone Debugger, which supports both loading and debugging. Since you have your own PCB, you will most likely build the firmware yourself.  In that case, you get as result of the build a firmware.elf file, which contains the code and symbols. But there will be also firmware.hex and firmware.bin files. There is a small utility by Segger called JFlashLite which you can use for flashing the firmware. The load address is 0x60000000. But that is only needed for the .bin file. The .hex file has the flashing address encoded.
